wolfcrypt/src/hash.c

```diff
@@ -840,6 +840,7 @@ int wc_HashFree(wc_HashAlg* hash, enum wc_HashType type)
             }
             wc_Sha256Free(sha256);
         }
+	wc_Sha256Free(sha256);
```

Comment:
Why are 2 Sha256Frees needed 2 lines apart?

Reply:
One was if Init succeeded and the other if it failed. It's wrong to need it if Init failed, am updating the wc_InitSha256 function to fix this.